tests/cprFullNameGenderTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+require_once 'src/FakeInfo.php';
+
+
+use PHPUnit\Framework\TestCase;
+
+class TestgetCprFullNameGender extends testCase {
+
+    private FakeInfo $testInfo;
+    
+    public function setUp(): void {
+        $this->fakeInfo = FakeInfo;
+    }
+    public function tearDown(): void {
+        unset($this->fakeinfo);
+    }
+    
+    // Test we wanna do is 
+    //  - Check if we get a array
+    //  - Check if array is empty
+    //  - Check if the array contains the number of indexs we expected
+    //  - Check if all the indexs are there 
+
+    public function testIfArray() {
+        $result = is_array($this->fakeInfo->getCprFullNameAndGender());
+        $expected = "true";
+
+        $this->assertEquals($result, $expected);
+    }
+
+    public function testIfEmpty() {
+        $result = empty($this->fakeInfo->getCprFullNameAndGender());
+        $expected = "false";
+
+        $this->assertEquals($result, $expected);
+    }
+
+    public function testArraySize() {
+        $result = sizeof($this->fakeInfo->getCprFullNameAndGender());
+        $expected = 4;
+
+        $this->assertEquals($result, $expected);
+    }
+
+    // Testing if the keys we want are there
+
+    // CPR
+    public function testCprKey() {
+        $result = in_array('CPR', $this->fakeInfo->getCprFullNameAndGender());
+        $expected = "true";
+
+        $this->assertEquals($result, $expected);
+    }
+
+    // firstName
+    public function testFirstNameKey() {
+        $result = in_array('firstName', $this->fakeInfo->getCprFullNameAndGender());
+        $expected = "true";
+
+        $this->assertEquals($result, $expected);
+    }
+
+    // lastName
+    public function testLastNameKey() {
+        $result = in_array('lastName', $this->fakeInfo->getCprFullNameAndGender());
+        $expected = "true";
+
+        $this->assertEquals($result, $expected);
+    }
+
+    // gender
+    public function testGenderKey() {
+        $result = in_array('gender', $this->fakeInfo->getCprFullNameAndGender());
+        $expected = "true";
+
+        $this->assertEquals($result, $expected);
+    }
+
```

Comment:
Som i den anden kan det overvejes om det ville være smart at skrive én samlet test af nøglerne med en dataprovider.

Reply:
Har prøvet at tjekke dataprovider og jeg kunne ikke få mit hoved omkring det helt, men jeg har gjordt resten tak for hjælpen ^^

Reply:
Du sætter en dataprovider op ved at placere sådan en type meta kommentar direkte over din test funktion:
```
  /**
   * @dataProvider navnetpådindataprovider
   */
```
Og så laver du en funktion med samme navn som returnerer et array med de data du vil teste - fx navnene på array keys som vi forventer 
```
    public function navnetpådindataprovider() {
        return [
            ['CPR'], 
            ['firstName'], 
            ['lastName'], 
            ['gender'], 
        ];
    }
```
Og i din funktion så indsætter du det som et parameter 
```
    public function testname($key) {
        $result = in_array($key, $this->fakeInfo->getCprFullNameAndGender());
        $this->assertTrue($result);
    }
```

Reply:
Okay jeg har sat dataproviders ind på begge, og har pushede det